src/shared/contexts/UserPreferences.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  createContext,
+  useContext,
+  useCallback,
+  useEffect,
+  FC,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { frame, schedule } from 'timing-functions';
+import { JsonValue } from 'type-fest';
+
+import useSafeState from '../hooks/useSafeState';
+import { Namespace } from '../types/namespaces';
+
+type UserPreferences = Record<string, JsonValue>;
+
+type ContextType = [
+  state: UserPreferences,
+  setState: Dispatch<SetStateAction<UserPreferences>>
+];
+
+const unusedDefaults: ContextType = [
+  {}, // will be global state of all user preferences hooks
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  () => {}, // will be global state setter
+];
+const UserPreferencesContext = createContext<ContextType>(unusedDefaults);
+
+UserPreferencesContext.displayName = 'UserPreferencesContext';
+
+export const UserPreferencesProvider: FC = ({ children }) => {
+  // Runtime state for all user preferences, lazily populated with usage.
+  // Using useSafeState because we *might* set values in the global state after
+  // the component has unmounted, it's OK, but would trigger a warning otherwise
+  const stateAndSetState = useSafeState<UserPreferences>({});
+
+  return (
+    <UserPreferencesContext.Provider value={stateAndSetState}>
+      {children}
+    </UserPreferencesContext.Provider>
+  );
+};
+
+export type PossibleUserPreferenceKey =
```

Comment:
I think `UserPreferenceKey` would be sufficient. Also thinking this might be better situated somewhere in src/app as we currently do for the Locations.

Reply:
Well, I'm exporting it, but there's actually no need to export it so it just needs to be defined within this file. You get autocomplete wherever you use the hook.
I will still change the name though